instrumentation/kafka/kafka-connect-2.6/testing/src/test/java/io/opentelemetry/instrumentation/kafkaconnect/v2_6/MongoKafkaConnectSinkTaskTest.java

```
@@ -0,0 +1,352 @@
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.restassured.http.ContentType;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.bson.Document;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers
+class MongoKafkaConnectSinkTaskTest extends KafkaConnectSinkTaskBaseTest {
+  // MongoDB-specific constants
+  private static final String MONGO_NETWORK_ALIAS = "mongodb";
+  private static final String DB_NAME = "testdb";
+  private static final String COLLECTION_NAME = "person";
+  private static final String CONNECTOR_NAME = "test-mongo-connector";
+  private static final String TOPIC_NAME = "test-mongo-topic";
+
+  private static MongoDBContainer mongoDB;
+
+  @Override
+  protected void setupDatabaseContainer() {
+    mongoDB =
+        new MongoDBContainer(DockerImageName.parse("mongo:4.4"))
+            .withNetwork(network)
+            .withNetworkAliases(MONGO_NETWORK_ALIAS)
+            .withStartupTimeout(Duration.ofMinutes(5));
+  }
+
+  @Override
+  protected void startDatabaseContainer() {
+    Startables.deepStart(mongoDB).join();
+  }
+
+  @Override
+  protected void stopDatabaseContainer() {
+    if (mongoDB != null) {
+      try {
+        mongoDB.stop();
+      } catch (RuntimeException e) {
+        logger.error("Error stopping MongoDB: " + e.getMessage());
+      }
```

Usually we just let the errors propagate. No need to catch anything unless you expect it to fail and need to catch the exception to avoid it breaking stuff.

---

I removed the catch blocks but left them around handful of places as compiler was complaining about lack of error handling.